### ISSUE TITLE
x:param overriding global xsl:param not working as documented

### ISSUE THREAD

**Issue by dariok** [xspec/xspec#1564]
I've encountered 2 issues with global params, working out of oXygen 24.0.

I'll be referring to [1] https://github.com/xspec/xspec/wiki/External-Transformation#xspec-global-parameters-scenario-level-parameters-and-global-variables and [2] https://github.com/xspec/xspec/wiki/Writing-Scenarios#global-parameters in the following lines.

1. **overriding global params**:
    Overriding a global `xsl:param` at scenario level only works if `x:param` is present globally, too. [2] says that you can put `x:param` globally or on scenario level. It is not clear whether this is a bug in the code or a restriction and just needs to be addressed in the documentation
1. **placement of scenario-level x:param**:
    scenario-level `x:param` needs to be a child of `x:context`, not `x:scenario`. [1] refers to scenario-level `x:param` as `//x:scenario/x:param`; validation in oXygen marks this as an error and it also does not work. It seems that `x:param` must be a child of `x:context` in order for overrides to work as expected.

Examples

Stylesheet:
```
<xsl:stylesheet xmlns:xsl="http://www.w3.org/1999/XSL/Transform" version="3.0">
   
   <xsl:param name="mode" />
   
   <xsl:template match="*:p">
      <p mode="{$mode}" />
   </xsl:template>
</xsl:stylesheet>
```

Test 1 (`x:scenario/x:param` as in [1]; failing):
```
<?xml version="1.0" encoding="UTF-8"?>
<x:description xmlns:x="http://www.jenitennison.com/xslt/xspec" stylesheet="Untitled.xsl" run-as="external">
   <x:scenario label="T1">
      <x:param name="mode">test</x:param>
      <x:context>
         <p></p>
      </x:context>
      <x:expect label=".">
         <p mode="test" />
      </x:expect>
   </x:scenario>
</x:description>
```

Test 2 (`x:context/x:param`; failing):
```
<x:description xmlns:x="http://www.jenitennison.com/xslt/xspec" stylesheet="Untitled.xsl" run-as="external">
   <x:scenario label="T1">
      <x:context>
         <x:param name="mode">test</x:param>
         <p></p>
      </x:context>
      <x:expect label=".">
         <p mode="test" />
      </x:expect>
   </x:scenario>
</x:description>
```

Test 3 (global `x:param` **and** `x:context/x:param`; passing):
```
<x:description xmlns:x="http://www.jenitennison.com/xslt/xspec" stylesheet="Untitled.xsl" run-as="external">
   <x:param name="mode"/>
   <x:scenario label="T1">
      <x:context>
         <x:param name="mode">test</x:param>
         <p></p>
      </x:context>
      <x:expect label=".">
         <p mode="test" />
      </x:expect>
   </x:scenario>
</x:description>
```

**Comment by AirQuick** [comment]
> [1] refers to scenario-level `x:param` as `//x:scenario/x:param`; validation in oXygen marks this as an error

Not reproduced on my end. Oxygen 24.0 build 2021101506 marks it as valid.
![valid](https://user-images.githubusercontent.com/8489367/141491373-80df92dc-4945-4edb-8def-a91f087bad1f.png)

> Test 1 (`x:scenario/x:param` as in [1]; failing):

Not reproduced on my end. Test 1 is passing.

> Test 2 (`x:context/x:param`; failing):

Reproduced on my end. Test 2 is failing.
It's working as expected, because `x:context/x:param` should take effect on `xsl:template/xsl:param` but the tested stylesheet has no such `xsl:param`.

> Test 3 (global `x:param` **and** `x:context/x:param`; passing):

Reproduced on my end. Test 3 is passing.
It is not an expected behavior. Looks like a bug in localizing `x:context/x:param`.

**Comment by dariok** [comment]
That is strange. I'm on the same build but get a validation error:
![Screenshot_20211113_192406](https://user-images.githubusercontent.com/1765292/141655019-e7c799d4-0eee-4892-a7e7-387440fc13e2.png)

And the result:

![Screenshot_20211113_192706](https://user-images.githubusercontent.com/1765292/141655105-18de39c9-5586-4e12-ab98-9cc72b71a602.png)

**Comment by AirQuick** [comment]
Can you double click the validation error message and see where the associated schema comes from?

https://user-images.githubusercontent.com/8489367/141681959-a30f26fd-fd70-4b4d-aa09-fb46a208f610.mp4

**Comment by AirQuick** [comment]
> Reproduced on my end. Test 3 is passing.
> It is not an expected behavior. Looks like a bug in localizing `x:context/x:param`.

Will be fixed by #1574

**Comment by dariok** [comment]
Sorry for the late reply!

The schema used is
/home/dario/.com.oxygenxml/extensions/v24.0/frameworks/https_www.oxygenxml.com_InstData_Addons_community_updateSite.xml/xspec.support-2.0.0/src/schemas/xspec.rnc

This is automatically selected by oXygen and also used when explicitly using the “XSpec” validation scenario.

The schema in your example is present, too, but not used by oXygen. I can confirm, though, that the file validates against this one.

--

Given the different paths, I suspect the add-ins “XSpec Helper view” and its associate “XSpec Framework” as the culprits: I’ve just removed these two and now Test 1 does pass.

**Comment by AirQuick** [comment]
So it comes from [**Oxygen XSpec Helper View** add-on](https://github.com/xspec/oXygen-XML-editor-xspec-support). At this moment, the add-on bundles XSpec v2.0.7. Scenario-level `x:param` was first implemented on XSpec v2.1.2.

@xspec/oxygen-xml-editor-xspec-support Can you consider releasing a new version?

**Comment by AlexJitianu** [comment]
I've [added an issue](https://github.com/xspec/oXygen-XML-editor-xspec-support/issues/48) for this and I'll try to put it in the next sprint.
